src/main/java/nextstep/qna/domain/Question.java

```diff
@@ -81,12 +82,43 @@ public boolean isDeleted() {
         return deleted;
     }
 
-    public List<Answer> getAnswers() {
+    public Answers getAnswers() {
         return answers;
     }
 
     @Override
     public String toString() {
         return "Question [id=" + getId() + ", title=" + title + ", contents=" + contents + ", writer=" + writer + "]";
     }
+
+    public void delete(NsUser loginUser) throws CannotDeleteException {
+        validateOwner(loginUser);
+
+        if (this.answers.hasAnswer()) {
+            this.answers.delete(loginUser);
+        }
+
+        this.setDeleted(true);
```

Comment:
```suggestion
        DeleteHistoryTargets history = this.answers.delete(loginUser);
```

이부분도 메세지를 보내서 처리하는게 좋아보여요!

Reply:
이 부분도 위와 동일한 맥락으로, 변경하였습니다~!